### PR TITLE
GG-22147: SQL: Create a view with list of existing indexes

### PR DESCRIPTION
**Pull request by ygerzhedovich**
GG-22147: SQL: Create a view with list of existing indexes

### CODE REVIEW

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java [hunk 1]
```diff
@@ -154,6 +156,113 @@ public void testModifications() throws Exception {
...
+
+        String idxSql = "SELECT * FROM IGNITE.INDEXES ORDER BY TABLE_NAME, INDEX_NAME";
+
+        List<List<?>> clientNodeNodeIndexes = execSql(client, idxSql);
+
+        assertTrue("Only affinity nodes know about their indexes", clientNodeNodeIndexes.isEmpty());
+
+        List<List<?>> srvNodeIndexes = execSql(srv, idxSql);
+
+        String[][] expectedResults = {
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java]: abbreviation should be used

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java [hunk 2]
```diff
@@ -154,6 +156,113 @@ public void testModifications() throws Exception {
...
+
+        execSql("CREATE INDEX IDX_3 ON PUBLIC.DFLT_CACHE(MY_VAL)");
+
+        execSql("CREATE TABLE PUBLIC.DFLT_AFF_CACHE (ID1 INT, ID2 INT, MY_VAL VARCHAR, PRIMARY KEY (ID1 DESC, ID2)) WITH \"affinity_key=ID1\"");
+
+        execSql("CREATE INDEX IDX_AFF_1 ON PUBLIC.DFLT_AFF_CACHE(ID2 DESC, ID1, MY_VAL DESC)");
+
+        String idxSql = "SELECT * FROM IGNITE.INDEXES ORDER BY TABLE_NAME, INDEX_NAME";
+
+        List<List<?>> clientNodeNodeIndexes = execSql(client, idxSql);
```
Comment (korlov42) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java]: extra "Node" in the name.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+import org.h2.engine.Session;
+import org.h2.result.Row;
+import org.h2.result.SearchRow;
+import org.h2.value.Value;
+
+/**
+ * View that contains information about all the sql tables in the cluster.
+ */
+public class SqlSystemViewIndexes extends SqlAbstractLocalSystemView {
+
```
Comment (korlov42) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java]: empty line

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java [hunk 4]
```diff
@@ -210,6 +214,56 @@ public GridH2Table(CreateTableData createTblData, GridH2RowDescriptor desc, H2Ro
...
+    @SuppressWarnings("ZeroLengthArrayAllocation")
+    public List<IndexInformation> indexesInformation() {
+        List<IndexInformation> res = new ArrayList<>();
+
+        IndexColumn keyCol = indexColumn(KEY_COL, SortOrder.ASCENDING);
+
+        List<IndexColumn> wrappedKeyCols = H2Utils.treeIndexColumns(rowDescriptor(),
+            new ArrayList<>(2), keyCol, affKeyCol);
+
+        //explicit add HASH index, due to we know all their parameters and it doesn't created on non afinity nodes.
```
Comment (korlov42) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java]: there should be space after "//". Also fix line 237.

Comment (korlov42) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java]: typo in the word "affinity".

Comment (ygerzhedovich) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/opt/GridH2Table.java]: fixed.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java (hunk 3 above)
Comment (ygerzhedovich) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java]: fixed.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java (hunk 2 above)
Comment (ygerzhedovich) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java]: fixed.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2IndexType.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.ignite.internal.processors.query.h2.database;
+
+/**
+ * Type of supported indexed types.
```
Comment (korlov42) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2IndexType.java]: I suppose there should be "indexes" not "indexed".

File: modules/geospatial/src/test/java/org/apache/ignite/internal/processors/query/h2/H2IndexesSystemViewTest.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Test expose SPATIAL indexes through SQL system view INDEXES.
+ */
+public class H2IndexesSystemViewTest extends GridCommonAbstractTest {
+
```
Comment (korlov42) [inline modules/geospatial/src/test/java/org/apache/ignite/internal/processors/query/h2/H2IndexesSystemViewTest.java]: empty line

File: modules/geospatial/src/test/java/org/apache/ignite/internal/processors/query/h2/H2IndexesSystemViewTest.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        IgniteEx client = startGrid(getConfiguration().setClientMode(true).setIgniteInstanceName("CLIENT"));
+
+        execSql("CREATE TABLE PUBLIC.AFF_CACHE (ID1 INT, ID2 INT, GEOM GEOMETRY, PRIMARY KEY (ID1))");
+
+        execSql("CREATE SPATIAL INDEX IDX_GEO_1 ON PUBLIC.AFF_CACHE(GEOM)");
+
+        String idxSql = "SELECT * FROM IGNITE.INDEXES ORDER BY TABLE_NAME, INDEX_NAME";
+
+        List<List<?>> clientNodeNodeIndexes = execSql(client, idxSql);
```
Comment (korlov42) [inline modules/geospatial/src/test/java/org/apache/ignite/internal/processors/query/h2/H2IndexesSystemViewTest.java]: extra "Node"

File: modules/geospatial/src/test/java/org/apache/ignite/internal/processors/query/h2/H2IndexesSystemViewTest.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+        Assert.assertTrue("Only affinity nodes know about their indexes", clientNodeNodeIndexes.isEmpty());
+
+        List<List<?>> srvNodeIndexes = execSql(srv, idxSql);
+
+        String[][] expectedResults = {
```
Comment (korlov42) [inline modules/geospatial/src/test/java/org/apache/ignite/internal/processors/query/h2/H2IndexesSystemViewTest.java]: abbreviation should be used.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+ * View that contains information about all the sql tables in the cluster.
+ */
+public class SqlSystemViewIndexes extends SqlAbstractLocalSystemView {
+    /** H2 Indexing. */
+    private final IgniteH2Indexing idx;
+
+    /**
+     * Creates view with columns.
+     *
+     * @param ctx kernal context.
```
Comment (AMashenkov) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java]: Param idx is missed.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java [hunk 10]
```diff
@@ -154,6 +156,113 @@ public void testModifications() throws Exception {
...
+
+            assertEquals(expRow.length, resRow.size());
+
+            for (int j = 0; j < expRow.length; j++)
+                assertEquals(expRow[j], String.valueOf(resRow.get(j)));
+        }
+    }
+
+    /**
+     * @return Default cache configuration.
```
Comment (AMashenkov) [inline modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java]: Param cache name is missed.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2IndexType.java (hunk 5 above)
Comment (ygerzhedovich) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2IndexType.java]: Fixed.

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java (hunk 9 above)
Comment (ygerzhedovich) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/sys/view/SqlSystemViewIndexes.java]: Fixed.

File: modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java (hunk 10 above)
Comment (ygerzhedovich) [reply modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/SqlSystemViewsSelfTest.java]: Fixed.

**Pull request merged**